src/boa.ts

```diff
@@ -295,8 +295,13 @@ export function downloadOutput(uri: vscode.Uri|BoaJob) {
     runBoaCommands(async (client: boaapi.BoaClient) => {
         const jobId = getJobId(uri);
         const job = await client.getJob(jobId);
-        await job.downloadOutput(getWorkspaceRoot() + `/boa-job${jobId}-output.txt`);
-        vscode.window.showInformationMessage(`Output for Job ${jobId} downloaded`);
+
+        if (!job.running && await job.outputSize > 0 && job.compilerErrors == undefined) {
+            await job.downloadOutput(getWorkspaceRoot() + `/boa-job${jobId}-output.txt`);
+            vscode.window.showInformationMessage(`Output for Job ${jobId} downloaded`);
+        } else {
+            vscode.window.showInformationMessage(`Job ${jobId} is running`);
```

Comment:
Given the many ways a job might fail to have output, this could be made more generic.

```suggestion
            vscode.window.showInformationMessage(`Job ${jobId} has no output`);
```